controllers/client/grafana_client.go

```diff
@@ -36,19 +36,33 @@ func getAdminCredentials(ctx context.Context, c client.Client, grafana *v1beta1.
 			return credentials, nil
 		}
 
-		// rely on username and password otherwise
-		username, err := GetValueFromSecretKey(ctx, grafana.Spec.External.AdminUser, c, grafana.Namespace)
-		if err != nil {
-			return nil, err
+		switch {
+		case grafana.Spec.External.AdminUser != nil:
```

Comment:
Let's drop the change from this PR. - I feel like it's better to consider it separately + cover it with tests.